framework/src/util/LDAPExpr.cpp

```diff
@@ -709,14 +709,23 @@ std::string LDAPExpr::ParseState::getAttributeName()
 
 std::string LDAPExpr::ParseState::getAttributeValue()
```

Comment:
should getAttributeValue and getAttributeName be implemented with std::regex? I know they can be a pain, but they might be simpler and more expressive than doing character by character parsing.

Reply:
I agree that we can look into re-implementing some parts of LDAPExpr/LDAPFilter.
Jeff and I briefly talked about this too, perhaps even using a third-party tool.

I think that work would be beyond the scope of this particular issue with parentheses, so I'll create a new issue to consider re-implementing LDAPFilter/LDAPExpr.